src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -70,26 +70,35 @@ static Oid ExtractFirstDistributedTableId(Query *query);
 static Const * ExtractInsertPartitionValue(Query *query, Var *partitionColumn);
 static Task * RouterSelectTask(Query *query);
 static Job * RouterQueryJob(Query *query, Task *task);
+static bool MultiRouterPlannableQuery(Query *query, MultiExecutorType taskExecutorType);
 static bool ColumnMatchExpressionAtTopLevelConjunction(Node *node, Var *column);
 static void SetRangeTablesInherited(Query *query);
 
 
 /*
- * MultiRouterPlanCreate creates a physical plan for given router plannable query.
- * Created plan is either a modify task that changes a single shard, or a router task
- * that returns query results from a single shard. Supported modify queries
- * (insert/update/delete) are router plannble by default. The caller is expected to call
- * MultiRouterPlannableQuery to see if the query is router plannable for select queries.
+ * MultiRouterPlanCreate creates a physical plan for given query. Created plan is
+ * either a modify task that changes a single shard, or a router task that returns
+ * query results from a single shard. Supported modify queries (insert/update/delete)
+ * are router plannable by default. If query is not router plannable then the function
+ * returns NULL.
  */
 MultiPlan *
-MultiRouterPlanCreate(Query *query)
+MultiRouterPlanCreate(Query *query, MultiExecutorType taskExecutorType)
```

Comment:
To what end? Now it had to be visible to multi planner.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.